CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+
+When you make a pull request that fixes an issue, include a line that says `Fixes #X` (or "Closes"), where `X` is the issue number.
+This will cause the issue in question to be closed when your PR is merged.
+
+General improvements to code quality are also welcome!
+Bevy can always be safer, better tested, and more idiomatic.
+
+### Writing docs
+
+Like every other large, rapidly developing open source library you've ever used, Bevy's documentation could stand to be improved.
+This is incredibly valuable, easily distributed work, but requires a bit of guidance:
+
+* inaccurate documentation is worse than no documentation: prioritize fixing broken docs
+* Bevy is remarkable unstable: before tackling a new major documentation project, check in with the community about the stability of that feature and upcoming plans to save yourself heartache
+* code documentation (doc examples and in the examples folder) is easier to maintain because the compiler will tell us when it breaks
+* inline documentation should be fairly direct, minimal and technical
+* the Bevy book is hosted on the `bevy-website` repo and targeted towards beginners who are just getting to know Bevy (and perhaps Rust!)
+* RFCs are not documentation: they serve as a record of accepted decisions
+
+[docs.rs](https://docs.rs/bevy) is built out of the last release's documentation, which is written right beside the adjacent code.
+To view the current docs on `main` before you contribute, clone the `bevy` repo, and run `cargo doc --open`.
+
+### Writing examples
+
+Most [examples in Bevy](https://github.com/bevyengine/bevy/tree/main/examples) aim to clearly demonstrate a single feature, group of closely related small features, or show how to accomplish a particular task (such as asset loading, creating a custom shader or testing of your app).
+In rare cases, creating new "game" examples is justified to demonstrate new features
+that open a complex class of functionality in a way that's hard to showcase in isolation or requires additional integration testing.
+
+Examples in Bevy should be:
+
+1. **Working.** They must compile and run, and should fail in obvious ways.
+2. **Clear.** They must use descriptive variable names, have reasonable code-quality, be formatted, and be appropriately commented.
+3. **Relevant.** They should use explain, through comments or variable names, how what they're demonstrating could be concretely useful to a game developer.
+4. **Minimal.** They should be no larger or more complex than is needed to meet their other goals.This reduces maintenance burden and improves clarity when used as a reference.
+
+When you add a new example, be sure to update `examples/README.md` with the new example.
+Use a generous sprinkling of keywords in your description: these are commonly used to locate a specific example using CTRL+F.
```

Comment:
nit: "using CTRL+F" doesn't work on mac. of course every mac user is used to translate it directly but saying "when searching" achieve the same goal and is nicer imho